kubeflow/trainer/rhai/utils.py

```
@@ -30,8 +36,99 @@ def get_trainer_cr_from_rhai_trainer(
         )
 
     else:
-        _raise_unknown_rhai_trainer(trainer)
+        raise ValueError(f"Unknown trainer {trainer}.")
```

Ruff (0.14.5)

Prefer `TypeError` exception for invalid type

(TRY004)